src/kOS/Execution/CPU.cs

```diff
@@ -1031,7 +1037,12 @@ private void ProcessTriggers()
                 }
             }
 
-            currentContext.InstructionPointer = currentInstructionPointer;
+            // since `run` opcodes don't work in triggers, we can use the opcode count to determine if the
+            // program has been aborted.  If the count isn't right, leave the pointer where it is.
```

Comment:
It's been a while since I wrote this part.  I will have to review to see if there's a good way to get the Abort info back out of `ExecuteInstruction` As implemented, we just get the binary "continue or not" return value.  The problem is that right now the program has already been set back up after the execution of the reboot call opcode.  So If we reset the instruction pointer, it causes a misalignment.  Maybe we could switch to returning an enum instead of a bool.

For what it's worth, I'm not a fan of `run` working inside of triggers.  Interrupt and event driven programming should avoid long and/or complicated execution if possible, and I would say running a new program is by definition complicated.  But I won't argue with you on the feature.  I know that in my early kerbscript days I tried to do it, but that was before user functions.  I'm sure it would be useful to some one else, so leaving the door open for the feature is a good idea.